metrics/web/lib/dashboard/presentation/widgets/build_result_bar_component.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:metrics/dashboard/presentation/widgets/strategy/build_result_bar_padding_strategy.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+/// A widget that displays the [BuildResultBar], a corresponding [GraphIndicator]
+/// and a [BuildResultPopupCard].
+class BuildResultBarComponent extends StatelessWidget {
+  /// A width of the [BuildResultPopupCard] this bar component displays
+  /// in a popup.
+  static const double _popupWidth = 162.0;
+
+  /// A [BuildResultViewModel] with data to display.
+  final BuildResultViewModel buildResult;
+
+  /// A class that provides an [EdgeInsets] padding to apply to this bar based
+  /// on the [buildResult] position among other results.
+  final BuildResultBarPaddingStrategy paddingStrategy;
+
+  /// Creates a new instance of the [BuildResultBarComponent] with the given
+  /// [buildResult] and [paddingStrategy].
+  const BuildResultBarComponent({
+    Key key,
+    this.paddingStrategy,
+    this.buildResult,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    const barWidth = DimensionsConfig.graphBarWidth;
+    const indicatorDiameter = DimensionsConfig.graphIndicatorOuterDiameter;
+    const indicatorRadius = indicatorDiameter / 2.0;
+    const indicatorHorizontalOffset = (barWidth - indicatorDiameter) / 2.0;
+
+    if (buildResult == null) {
+      final inactiveTheme = MetricsTheme.of(context).inactiveWidgetTheme;
+
+      return PlaceholderBar(
+        width: barWidth,
+        height: 4.0,
+        color: inactiveTheme.primaryColor,
+      );
+    }
+
+    return BasePopup(
+      popupOpaque: false,
+      closeOnTapOutside: false,
+      popupConstraints: const BoxConstraints(
+        minWidth: _popupWidth,
+        maxWidth: _popupWidth,
+      ),
+      offsetBuilder: (triggerSize) => _calculatePopupOffset(
+        triggerSize,
+        indicatorRadius,
+      ),
+      triggerBuilder: (context, openPopup, closePopup, isOpened) {
+        return MouseRegion(
+          cursor: SystemMouseCursors.click,
+          onEnter: (_) => openPopup(),
+          onExit: (_) => closePopup(),
+          child: GestureDetector(
+            behavior: HitTestBehavior.opaque,
+            onTap: _onTap,
+            child: Padding(
+              padding: _barPadding,
+              child: Stack(
+                clipBehavior: Clip.none,
+                fit: StackFit.passthrough,
+                children: [
+                  BuildResultBar(
+                    buildResult: buildResult,
+                  ),
+                  if (isOpened)
+                    Positioned(
+                      right: indicatorHorizontalOffset,
+                      bottom: -indicatorRadius,
+                      child: _graphIndicator,
+                    ),
+                ],
+              ),
+            ),
+          ),
+        );
+      },
+      popup: BuildResultPopupCard(
```

Comment:
Would it be a bit more readable if we'll move this parameter to be before `triggerBuilder`?